request.js

```diff
@@ -92,7 +92,19 @@ function request(requestOptions, callback) {
   debug(requestId, "calledBack:", calledBack)
   var req = transport.request(options, function onResponse(res) {
     metadata = res
-    res.pipe(concat(collect))
+
+    var contentLength
+    try {
+      contentLength = parseInt(res.headers['content-length'], 10)
```

Comment:
Good point, I think it's likely enough that no `Content-Length` header would be sent, but in that case it should probably be assumed that there will probably be content, it's just of indeterminate length. (Such a scenario is common with streaming endpoints, not that this library is designed to support streaming endpoints.)